### PR TITLE
VMware Plugin: Adapt to Python 3.12

### PR DESCRIPTION
**Pull request by sduehr**
The changed configparser and ssl modules in Python 3.12 required adaptions of the plugin code so that it now works with the Python versions 3.6, 3.9, 3.10 and 3.12

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

### CODE REVIEW

**Comment by sebsura** [comment]
Would it also be possible to fix these error messages ?
```
bareos/core/cmake/get_python_compile_settings.py:37: SyntaxWarning: invalid escape sequence '\ '
  'message(STATUS "Python{0}_{1}\ is\  {2}")'.format(
bareos/core/cmake/get_python_compile_settings.py:50: SyntaxWarning: invalid escape sequence '\ '
  'message(STATUS "Python{0}_{1}_FLAGS\ is\  {2}")'.format(
bareos/core/cmake/get_python_compile_settings.py:59: SyntaxWarning: invalid escape sequence '\ '
  'message(STATUS "Python{0}_{1}\ is\  {2}")'.format(
bareos/core/cmake/get_python_compile_settings.py:70: SyntaxWarning: invalid escape sequence '\ '
  'message(STATUS "Python{0}_{1}\ is\  {2}")'.format(
```
Im not sure why we tried to espace spaces (i have found nothing online about this) but with python 3.12 it actually throws these warnings.  We should just replace `\ ` by ` ` i guess.

File: core/src/plugins/filed/python/vmware/bareos-fd-vmware.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def parse_config_file(self):
         self.config = configparser.ConfigParser()
 
         try:
-            self.config.readfp(open(self.options["config_file"]))
+            self.config.read_file(open(self.options["config_file"]))
```
Comment (arogge) [inline core/src/plugins/filed/python/vmware/bareos-fd-vmware.py:180]: this looks like `read_file()` would consumes the file object, which it doesn't.
Maybe a this is a bit more readable?
```suggestion
            with open(self.options["config_file"]) as fp:
                self.config.read_file(fp)
```

File: core/src/plugins/filed/python/vmware/bareos-fd-vmware.py [hunk 2]
```diff
@@ -2957,35 +2957,47 @@ def cleanup_tmp_files(self):
...
         bareosfd.DebugMessage(
             100,
-            "retrieve_vcthumbprint() Retrieving SSL ThumbPrint from %s\n"
+            "fetch_vcthumbprint() Retrieving SSL ThumbPrint from %s\n"
             % (self.options["vcserver"]),
         )
+
         try:
-            wrappedSocket.connect((self.options["vcserver"], 443))
+            sock = socket.create_connection((self.options["vcserver"], 443), timeout=5)
```
Comment (arogge) [inline core/src/plugins/filed/python/vmware/bareos-fd-vmware.py:2978]: this looks way more complicated than it actually is.
As we're not handling exceptions, but just reporting that they happened, we could just reduce everything to one single try/except block like this:
```
        try:
            with socket.create_connection((self.options["vcserver"], 443), timeout=5) as sock:
                with ssl_context.wrap_socket(
                    sock, server_hostname=self.options["vcserver"]
                ) as wrappedSocket:
                    der_cert_bin = wrappedSocket.getpeercert(True)
                    thumb_sha1 = hashlib.sha1(der_cert_bin).hexdigest()
                    self.fetched_vcthumbprint = thumb_sha1.upper()
            return True
        except Exception as e:
            bareosfd.JobMessage(
                bareosfd.M_FATAL,
               "Could not fetch SSL Cert from %s: %s\n"
                % (self.options["vcserver"], str(e)),
            )
            return False
```

**Review by arogge — CHANGES_REQUESTED** [review]
The code is a more complicated than it should be, I think we can simplify this.

File: core/src/plugins/filed/python/vmware/bareos-fd-vmware.py (hunk 1 above)
Comment (sduehr) [reply core/src/plugins/filed/python/vmware/bareos-fd-vmware.py]: Ok, using `with open(...)` is obviously better readable, while technically `read_file(open(...))` is ok too, it doesn't leave the file open, also appears like this in an example in the configparser docs.

File: core/src/plugins/filed/python/vmware/bareos-fd-vmware.py (hunk 2 above)
Comment (sduehr) [reply core/src/plugins/filed/python/vmware/bareos-fd-vmware.py]: Sure, that looks a lot better indeed. Thanks.

**Comment by sduehr** [comment]
> Would it also be possible to fix these error messages ?
> 
> ```
> bareos/core/cmake/get_python_compile_settings.py:37: SyntaxWarning: invalid escape sequence '\ '
>   'message(STATUS "Python{0}_{1}\ is\  {2}")'.format(
> bareos/core/cmake/get_python_compile_settings.py:50: SyntaxWarning: invalid escape sequence '\ '
>   'message(STATUS "Python{0}_{1}_FLAGS\ is\  {2}")'.format(
> bareos/core/cmake/get_python_compile_settings.py:59: SyntaxWarning: invalid escape sequence '\ '
>   'message(STATUS "Python{0}_{1}\ is\  {2}")'.format(
> bareos/core/cmake/get_python_compile_settings.py:70: SyntaxWarning: invalid escape sequence '\ '
>   'message(STATUS "Python{0}_{1}\ is\  {2}")'.format(
> ```
> 
> Im not sure why we tried to espace spaces (i have found nothing online about this) but with python 3.12 it actually throws these warnings. We should just replace `\ ` by ` ` i guess.

I changed it within the last commit.

File: core/src/plugins/filed/python/vmware/bareos-fd-vmware.py (hunk 1 above)
Comment (arogge) [reply core/src/plugins/filed/python/vmware/bareos-fd-vmware.py]: You're right, it isn't wrong. However, you have to know that a) the file is closed when the file-handle is deleted and b) ConfigParser.read_file() doesn't store the file-handle somewhere

**Review by arogge — APPROVED** [review]
Looks great!

**Comment by arogge** [comment]
You still need to re-run bareos-check-sources on `core/cmake/get_python_compile_settings.py` to update the copyright header.

**Pull request merged**